backend/redis/healthRedis.go

```diff
@@ -0,0 +1,56 @@
+package redis
+
+import (
+	"github.com/go-redis/redis"
+	"github.com/pace/bricks/maintenance/health/servicehealthcheck"
+	"sync"
+	"time"
+)
+
+type redisHealthCheck struct {
+	State  *servicehealthcheck.ConnectionState
+	client *redis.Client
+}
+
+func (h *redisHealthCheck) InitHealthCheck() error {
+	h.client = Client()
+	h.State = &servicehealthcheck.ConnectionState{M: sync.Mutex{}}
+	return nil
+}
+
+func (h *redisHealthCheck) Name() string {
+	return "redis"
+}
+
+// HealthCheck if the last result is outdated, redis is checked for writeability and readability,
+// otherwise return the old result
+func (h *redisHealthCheck) HealthCheck(currTime time.Time) error {
+	h.State.M.Lock()
```

Comment:
The state update and the lock around it should be part of the `h.State` and not leak to the users of the `h.State`